packages/mixin-get-child-by-name/test/index.js

```diff
@@ -23,13 +23,27 @@ describe('PIXI.Container#getChildByName', function ()
         expect(parent.getChildByName).to.be.function;
     });
 
-    it('should enable getChildByName', function ()
+    it('should correctly find a child by its name', function ()
     {
         const obj = new DisplayObject();
         const parent = new Container();
 
         obj.name = 'foo';
         parent.addChild(obj);
+
         expect(parent.getChildByName('foo')).to.equal(obj);
     });
+
+    it('should correctly find a indirect child by its name in recursive search', function ()
+    {
+        const obj = new DisplayObject();
+        const parent = new Container();
+        const grandParent = new Container();
+
+        obj.name = 'foo';
+        parent.addChild(obj);
+        grandParent.addChild(obj);
```

Comment:
Shouldn’t this be addChild(parent)?

Reply:
Also, I think there’s more test coverage we could be doing for this feature. For instance, if a child and a descendant both have the same name, the highest level in the hierarchy should return first.

Reply:
Tests should cover few more cases:
1) If you passed a non existing name (2 scenarios with is Recursive = true/false)
2) If you passed an existing name (2 scenarios with is Recursive = true/false)
3) Always return the firs it has found
Hope it helps